R/execution.R

```diff
@@ -745,6 +913,83 @@ writeJsonResultsToTable <- function(connectionDetails,
+                                       "UNIT_CONCEPT_ID", "NUM_VIOLATED_ROWS",
+                                       "PCT_VIOLATED_ROWS", "NUM_DENOMINATOR_ROWS",
+                                       "EXECUTION_TIME", "NOT_APPLICABLE_REASON",
+                                       "ERROR", "QUERY_TEXT"),
+                           delimiter = "\t") {
+  tryCatch(
+    expr = { 
+      ParallelLogger::logInfo(sprintf("Writing results to CSV file %s", csvPath))
+      columns <- intersect(union(c("checkId", "FAILED", "PASSED", "IS_ERROR", "NOT_APPLICABLE"), columns), colnames(checkResults))
+      if (is.element("QUERY_TEXT", columns)) {
+        checkResults$QUERY_TEXT <- stringr::str_replace_all(checkResults$QUERY_TEXT, "\n", " ")
+        checkResults$QUERY_TEXT <- stringr::str_replace_all(checkResults$QUERY_TEXT, "\r", " ")
+        checkResults$QUERY_TEXT <- stringr::str_replace_all(checkResults$QUERY_TEXT, "\t", " ")
+      }
+      if (is.element("ERROR", columns)) {
+        checkResults$ERROR <- stringr::str_replace_all(checkResults$ERROR, "\n", " ")
+        checkResults$ERROR <- stringr::str_replace_all(checkResults$ERROR, "\r", " ")
+        checkResults$ERROR <- stringr::str_replace_all(checkResults$ERROR, "\t", " ")
+      }
+      write.table(dplyr::select(checkResults, columns), file = csvPath, sep = delimiter, row.names = FALSE, na = "")
+      ParallelLogger::logInfo("Finished writing to CSV file")
+    },
+    error = function(e) {
+      ParallelLogger::logError(sprintf("Writing to CSV file failed: %s", e$message))
+    }
+  )
+}
+
+#' Write JSON Results to SQL Table
+#' 
+#' @param jsonPath    Path to the JSON results file generated using the execute function
+#' @param csvPath     Path to the CSV output file
+#' @param columns     (OPTIONAL) List of desired columns
+#' @param delimiter   (OPTIONAL) CSV delimiter
+#' 
+#' @export
+writeJsonResultsToCsv <- function(jsonPath,
+                                  csvPath,
+                                  columns = c("checkId", "FAILED", "PASSED", 
+                                              "IS_ERROR", "NOT_APPLICABLE",
+                                              "CHECK_NAME", "CHECK_DESCRIPTION",
+                                              "THRESHOLD_VALUE", "NOTES_VALUE",
+                                              "CHECK_LEVEL", "CATEGORY",
+                                              "SUBCATEGORY", "CONTEXT",
+                                              "CHECK_LEVEL", "CDM_TABLE_NAME",
+                                              "CDM_FIELD_NAME", "CONCEPT_ID",
+                                              "UNIT_CONCEPT_ID", "NUM_VIOLATED_ROWS",
+                                              "PCT_VIOLATED_ROWS", "NUM_DENOMINATOR_ROWS",
+                                              "EXECUTION_TIME", "NOT_APPLICABLE_REASON",
+                                              "ERROR", "QUERY_TEXT"),
+                                  delimiter = "\t") {
```

Comment:
Again, I would expect a comma as delimiter.
```suggestion
                                  delimiter = ",") {
```